### PR TITLE
Many improvements

### PR DESCRIPTION
**Pull request by Matzey**
First of all I changed UI elements, that were basically a decision of between two elements, that were of Type QLineEdit to QComboBox, giving the User a DropDown Menu. I also added (s)css for those. (Don't forget to recompile with sass).

I also changed (hopefully more or less correct and understandable) the UnitTests to ones that have mocked values, so that the GitHub Actions don't need environment variables anymore. For this I also needed to change some underlying functions, in a way that made the tests easier (or even possible in the first place) without sacrificing the functionality in any way. The UnitTests also required a new library (which can be found in the requirement.txt).

### CODE REVIEW

**Comment by K0ntr4** [comment]
Let's keep the input elements as QLineEdit to allow users to input custom text. The choices are more or less just recommendations for better image generation results.

**Review by K0ntr4 — CHANGES_REQUESTED** [review]
See comment

**Review by K0ntr4 — CHANGES_REQUESTED** [review]
Fix:
src/image/game_info.py:89:5 E303 too many blank lines (2) [pycodestyle]
src/tests/test_character_info.py:24:1 W293 blank line contains whitespace [pycodestyle]
src/tests/test_character_info.py:31:1 W293 blank line contains whitespace [pycodestyle]
src/tests/test_character_info.py:55:1 W293 blank line contains whitespace [pycodestyle]
src/tests/test_game_info.py:18:39 W291 trailing whitespace [pycodestyle]
src/tests/test_game_info.py:34:1 W293 blank line contains whitespace [pycodestyle]
src/tests/test_game_info.py:51:1 W293 blank line contains whitespace [pycodestyle]
src/tests/test_game_info.py:70:1 W293 blank line contains whitespace [pycodestyle]
src/tests/test_game_info.py:87:1 W293 blank line contains whitespace [pycodestyle]

File: src/tests/test_game_info.py [hunk 1]
```diff
@@ -1,51 +1,94 @@
...
+        Test IGDB Response on https://api.igdb.com/v4/genres
+        """
+        mock_response = Mock()
+        response_body = {
+            "mid": ["clan", "Experience", "epic", "conclusion", "reasons", "Fighting"]
+        }
+        mock_response.json.return_value = response_body
+        mock_get.return_value = mock_response
+        
         keywords = self.game_info.get_game_keywords('Tekken 7')
```
Comment (K0ntr4) [inline src/tests/test_game_info.py:88]: Fix:
ERROR: test_get_game_keywords (test_game_info.TestGameInfo)
Test IGDB Response on https://api.igdb.com/v4/genres
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.[14](https://github.com/K0ntr4/anigame_fusion/actions/runs/8775511787/job/24078752471?pr=3#step:5:15)/x64/lib/python3.10/unittest/mock.py", line 1379, in patched
    return func(*newargs, **newkeywargs)
  File "/home/runner/work/anigame_fusion/anigame_fusion/src/tests/test_game_info.py", line 88, in test_get_game_keywords
    keywords = self.game_info.get_game_keywords('Tekken 7')
  File "/home/runner/work/anigame_fusion/anigame_fusion/src/image/game_info.py", line 132, in get_game_keywords
    game_details = self.get_game_info(name)
  File "/home/runner/work/anigame_fusion/anigame_fusion/src/image/game_info.py", line 93, in get_game_info
    self.authenticate()
  File "/home/runner/work/anigame_fusion/anigame_fusion/src/image/game_info.py", line 83, in authenticate
    response.raise_for_status()
  File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/requests/models.py", line 10[21](https://github.com/K0ntr4/anigame_fusion/actions/runs/8775511787/job/24078752471?pr=3#step:5:22), in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://id.twitch.tv/oauth2/token

Comment (Matzey) [reply src/tests/test_game_info.py:98]: This error is cause by the old unit test failing, because the environment variables are missing now.

Comment (K0ntr4) [reply src/tests/test_game_info.py:98]: That's why you should fix it. get_game_keywords also calls authenticate function to get the needed authentication token. You can implement a similar approach as in test_get_genres to just set a token.

**Review by K0ntr4 — CHANGES_REQUESTED** [review]

**Review by K0ntr4 — CHANGES_REQUESTED** [review]
See conversation. nvm

**Review by K0ntr4 — APPROVED** [review]

**Pull request merged**